### PR TITLE
Wip

### PR DESCRIPTION
**Pull request by junglie85**
A bunch of refactoring and general tidying up. The code should be cleaner and better organised.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,5 +3,56 @@
 
 # TicTacToe (Ruby Edtion)
 
-The venerable game of TicTacToe, written in Ruby.
+The venerable game of TicTacToe, written in Ruby, otherwise known as noughts and crosses.
```
Comment (ssunday) [inline README.md:6]: TIL

File: README.md [hunk 2]
```diff
@@ -3,5 +3,56 @@
...
+```
+git clone git@github.com:AshleyByeUK/tictactoe-ruby.git
+cd tictactoe-ruby
+```
+
+All of the following commands assume the current working directory is `tictactoe-ruby`.
+
+## Playing
+
+Make sure that the game's entry point is executable:
```
Comment (ssunday) [inline README.md:26]: Git does check in the files permissions...so I think you can do some git magic to make sure this file is executable in VC.

File: README.md [hunk 3]
```diff
@@ -3,5 +3,56 @@
...
+
+Run the game (executable from anywhere - add the `bin` directory to your `PATH` if you want):
+
+```
+./bin/tictactoe
+```
+
+## Testing
+
+The tests can be run with `bundler`:
```
Comment (ssunday) [inline README.md:40]: It's not that bundler is running it, it's that it's just a way to make sure you're using the executable of that thing found in that app's bundle. https://stackoverflow.com/questions/6588674/what-does-bundle-exec-rake-mean

Like global vs local installs....so it's rspec

File: lib/console_client/console_io.rb [hunk 4]
```diff
@@ -22,21 +22,15 @@ def exit
...
-      @exit
-    end
-
-    def exit=(exit)
-      @exit = exit
-    end
-
     def get_input(valid_input, error_message = '', prompt = '')
-      begin
+      input = nil
```
Comment (ssunday) [inline lib/console_client/console_io.rb:22]: This feels like it's doing more than just get input. It's validating/displaying errors as well...Hmm.

Comment (junglie85) [reply lib/console_client/console_io.rb:22]: Would it be better named as something like `get_validated_input`? Or is the function doing too much? I figured that this set of actions is repeated several times in various places, so needs to be encapsulated in a method somewhere.

**Pull request merged**